src/simple.test.ts

```diff
@@ -4,30 +4,39 @@ import { SimpleMerkleTree } from './simple';
 import { toHex } from './bytes';
 import { InvalidArgumentError, InvariantError } from './utils/errors';
 
-const leaf = fc.uint8Array({ minLength: 32, maxLength: 32 }).map(toHex);
+// Use a mix of uint8array and hexstring to cover the Byteslike space
+const leaf = fc
+  .uint8Array({ minLength: 32, maxLength: 32 })
+  .chain(l => fc.oneof(fc.constant(l), fc.constant(toHex(l))));
```

Comment:
This part is important IMO.

Until now, we tested the SimpleMerkleTree with hex strings, when it supports BytesLike. This change makes the leaf one-of "uint8array" or "toHex(uint8array)". This means the input will include (mix) both format to fully test the BytesLike support.

Reply:
Shouldn't we do the same for the standard one?

Reply:
standard one is limited by leafEncoding. It leaves little room. We would do that for the 'uint256' part, but since everything gets into the standard hash before being put in the tree, I don't think its that critical to fuzz

Reply:
Fair. I wouldn't fuzz values larger than `bytes32`.

Reply:
There is a unit test that push 33bytes of zero, and triggers an error